docs/source/etcd-data-model.rst

```diff
@@ -152,25 +168,134 @@ The various properties in this object have the following meanings:
+  can be applied to production workloads.
+
+  If ``labels`` is missing, it is treated as if there was an empty dict.
 
 .. _security-profile-data:
 
 Security Profiles
 ~~~~~~~~~~~~~~~~~
 
-Each security profile is split up into two bits of data: 'rules' and 'tags'.
+Each security profile is split up into three bits of data: 'rules', 'tags'
+and 'labels'.
+
 The 'rules' are an ordered list of ACLs, specifying what should be done with
 specific kinds of IP traffic. Traffic that matches a set of rule criteria will
-be accepted or dropped, depending on the rule. The tags are a list of
-classifiers that apply to each endpoint in the profile. The purpose of the
-tags is to allow for rules in other policies to refer to profiles by name,
-rather than by membership.
+be accepted or dropped, depending on the rule.
+
+The 'tags' are a list of classifiers that apply to each endpoint in the
+profile. The purpose of the tags is to allow for rules in other policies to
+refer to profiles by name, rather than by membership.
 
-For each profile, the rules objects and tag objects are stored in different
+Finally, labels contains a JSON dict with a set of key/value labels (as
+described above).  The labels on a profile are inherited by all the endpoints
+that directly reference that profile and they can be used in selectors as
+if they were directly applied to the endpoint.  'labels' is optional.
+
+For each profile, the rules, tags and labels objects are stored in different
 keys, of the form::
 
     /calico/v1/policy/profile/<profile_id>/rules
     /calico/v1/policy/profile/<profile_id>/tags
+    /calico/v1/policy/profile/<profile_id>/labels
+
+
+.. _security-policy-data:
+
+Tiered security policy
+~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to directly-referenced security profiles, Calico supports an even
+richer security model that we call "tiered policy". Naturally enough, tiered
+policy consists of a series of explicitly ordered "tiers".  Tiers contain
+(explicitly ordered) "policies".  Each policy has a Boolean selector expression
+attached that decides whether it applies to a given endpoint. Selector
+expressions match against an endpoint's labels.
+
+Each tier might have a different owner; for (a hypothetical) example, an
+enterprise's NetSec team could install a global black/white list that comes
+before rules generated by a Calico plugin::
+
+    tier 1: global "netsec" rules
+        policy 1, all endpoints: <global blacklist>
+        policy 2, all endpoints: <global whitelist>
+        ...
+    tier 2: Calico plugin-defined rules
+        policy 1, role == "webserver" && deployment == "prod": <prod webserver rules>
+    tier 3: ...
+
+
+Each tier of policy must approve each packet by using the "next-tier" action
+in a rule before is passes to the next.
+
+Calico renders the security policy for each endpoint individually and only the
+policies that have matching selectors are rendered.  This ensures that the
```

Comment:
Perhaps instead: 'Calico only programs kernel (iptables) rules for the security policies and profiles that apply to local endpoints.  Therefore the number of rules on a given host is proportional to the number of local endpoints, not the total amount of policy.  If no policies ...'